src/lib/checksum.c

```diff
@@ -150,3 +150,47 @@ uint32_t pseudo_header_initial(const int8_t *buf, size_t len)
   }
   return 0xFFFF0001;
 }
+
+int startoffset[2];
+int *prepare_packet(int8_t *buf, size_t len)
```

Comment:
Use `uint8_t startoffset[2]` as the values within `startoffset` will fit in 1 byte I think.
I would make the function fail fast. Check IPv4 or IPv6, check proto is TCP or UDP, do stuff.